src/queue.hpp

```diff
@@ -174,9 +174,18 @@ struct cvk_command_queue : public _cl_command_queue,
 
     cvk_buffer* get_or_create_printf_buffer() {
         if (!m_printf_buffer) {
-            cl_int status;
-            m_printf_buffer = cvk_buffer::create(
-                context(), 0, config.printf_buffer_size, nullptr, &status);
+            cl_int status = CL_SUCCESS;
+            auto buff_size_prop_index =
+                m_context->get_property_index(CL_PRINTF_BUFFERSIZE_ARM);
+            if (buff_size_prop_index == -1) {
+                cvk_error_fn("Could not get printf buffer size");
+                status = CL_INVALID_BUFFER_SIZE;
+            }
+            auto all_props = m_context->properties();
+            auto buff_size = all_props[buff_size_prop_index];
```

Comment:
This is a duplicate of what is done in `src/program.cc`. Please do that in the context constructor and keep the buffer size to return it without having anything to compute.